app/src/main/java/de/rampro/activitydiary/ui/history/HistoryActivity.java

```diff
@@ -176,44 +170,80 @@ protected void onCreate(Bundle savedInstanceState) {
         handleIntent(getIntent());
     }
 
-    @Override
-    protected void onNewIntent(Intent intent) {
-        setIntent(intent);
-        handleIntent(intent);
-    }
+    protected QHandler mQHandler = new QHandler();
 
     private void handleIntent(Intent intent) {
+        String query = null;
         if (ActivityDiarySuggestionProvider.SEARCH_ACTIVITY.equals(intent.getAction())) {
-            String query = intent.getStringExtra(SearchManager.QUERY);
+            query = intent.getStringExtra(SearchManager.QUERY);
             Uri data = intent.getData();
-            if(data != null) {
+            if (data != null) {
                 long id = Long.decode(data.getLastPathSegment());
                 filterHistoryView(id);
             }
-        }else if (ActivityDiarySuggestionProvider.SEARCH_NOTE.equals(intent.getAction())) {
+        } else if (ActivityDiarySuggestionProvider.SEARCH_NOTE.equals(intent.getAction())) {
             Uri data = intent.getData();
-            if(data != null) {
-                String query = data.getLastPathSegment();
+            if (data != null) {
+                query = data.getLastPathSegment();
                 filterHistoryNotes(query);
             }
 
-        }else if (ActivityDiarySuggestionProvider.SEARCH_GLOBAL.equals(intent.getAction())) {
+        } else if (ActivityDiarySuggestionProvider.SEARCH_GLOBAL.equals(intent.getAction())) {
             Uri data = intent.getData();
-            if(data != null) {
-                String query = data.getLastPathSegment();
+            if (data != null) {
+                query = data.getLastPathSegment();
                 filterHistoryView(query);
             }
-        }else if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
-            String query = intent.getStringExtra(SearchManager.QUERY);
+        } else if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
+            query = intent.getStringExtra(SearchManager.QUERY);
             filterHistoryView(query);
         }
-            /* TODO: save recent query
-            SearchRecentSuggestions suggestions = new SearchRecentSuggestions(this,
-                    MySuggestionProvider.AUTHORITY, MySuggestionProvider.MODE);
-            suggestions.saveRecentQuery(query, null);
-*/
+
+        /*
+            if query was searched, then insert query into suggestion table
+         */
+        if (query != null) {
+            LocalDBHelper mOpenHelper = new LocalDBHelper(this);
```

Comment:
also here the direct access is dangerous, we should do it through the content provider.